cmake/jana_plugin.cmake

```diff
@@ -118,8 +122,11 @@ macro(plugin_glob_all _name)
 
     # But... GLOB here makes this file just hot pluggable
     file(GLOB LIB_SRC_FILES CONFIGURE_DEPENDS *.cc *.cpp *.c)
-    #file(GLOB PLUGIN_SRC_FILES CONFIGURE_DEPENDS ${_name}.cc)
-    file(GLOB PLUGIN_SRC_FILES CONFIGURE_DEPENDS *.cc *.cpp *.c)
+    if(${_name}_WITH_LIBRARY)
+        file(GLOB PLUGIN_SRC_FILES CONFIGURE_DEPENDS ${_name}.cc)
+    else()
+        file(GLOB PLUGIN_SRC_FILES CONFIGURE_DEPENDS *.cc *.cpp *.c)
```

Comment:
`${_name}.cc` is a subset of `*.cc`

Reply:
See line 152.

Reply:
I didn't realize that was in. Nevermind this then.

Reply:
We can still fix issues even if it is in.

Reply:
Would be silly of me to leave messages if it wasn't fixable.

I think, the logic was and still is a bit complicated. If I understand correctly, the reason to move implementation of plugins into second-level libraries is for reuse of factory definitions, right? What if we've marked all factory symbols as invisible (they don't need to be exported as calls are routed via vtable). Then we could, at least, go back to static linking.

Also I'm not fond of eicrecon_* cmake functions. Stuff should be either a plugin or a library. We can have a single helper function for making a JANA plugin, but all the rest can be normal cmake stuff operating on single target at a time.

I'm sure I'm missing some important technical obstacles, but these are my current thoughts.

Reply:
I don't like the plugin/factory separation either. But it's what we have and I've tried to rewrite it twice into one or the other and failed both times. It's getting closer to being where I want it to be, at least. I don't think there should be any directory that needs to have both a plugin and a library.

I don't like the CMake macros all that much either, but if we ask for people to copy all that stuff all the time whenever they create a plugin or a library, then I don't want to have to clean up unnecessary or incorrect target_link_libraries all over. I think it at least isolates everything and makes sure it's consistent.